app/src/main/java/org/p2p/wallet/bridge/EthereumTokensPollingService.kt

```diff
@@ -0,0 +1,41 @@
+package org.p2p.wallet.bridge
+
+import timber.log.Timber
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.job
+import kotlinx.coroutines.launch
+import org.p2p.core.common.di.AppScope
+import org.p2p.wallet.bridge.interactor.EthereumInteractor
+
+private const val DELAY_IN_MILLISECONDS = 10_000L
+private const val TAG = "EthereumTokensPollingService"
+
+class EthereumTokensPollingService(
+    private val ethereumInteractor: EthereumInteractor,
+    appScope: AppScope
+) : CoroutineScope {
+
+    init {
+        val job = appScope.coroutineContext.job
+        job.invokeOnCompletion { pollingJob?.cancel() }
+    }
+
+    override val coroutineContext: CoroutineContext = appScope.coroutineContext
+    private var pollingJob: Job? = null
+
+    fun start() {
+        pollingJob = launch {
```

Comment:
I think there should be only single launch in case the start called more then once
1. pollingJob?.cancel()
2. pollingJob = launch ...